### PR TITLE
Handle repo redirect

### PR DESCRIPTION
**Pull request by JuanitoFatas**
When repo changed to a new location, the response contains a "Moved Permanently" message
and the new repo location url, we follow the url and get the (file) contents to find changelog-like file.

### CODE REVIEW

File: lib/whatsnew/remote_files.rb [hunk 1]
```diff
@@ -44,19 +44,41 @@ def client
...
 
       def find_from_contents
-        if resources = client.contents(repo)
-          resources.find do |resource|
-            resource.name =~ FILES_TO_SEARCH_REGEXP
-          end
+        resources = client.contents(repo)
+
+        if resources.respond_to?(:message) && resources.message == "Moved Permanently" &&
+           new_location = resources.url
```
Comment (winston) [inline lib/whatsnew/remote_files.rb]: Indentation?

File: lib/whatsnew/remote_files.rb [hunk 2]
```diff
@@ -44,19 +44,41 @@ def client
...
+        http_get(normalize_uri(url))
+      end
+
+      def normalize_uri(url)
+        require "uri"
+        URI(url)
+      end
+
+      def http_get(uri)
+        Dish(JSON.parse(uri.read))
```
Comment (winston) [inline lib/whatsnew/remote_files.rb]: Would prefer not to pull in another dependency just for `resources.name`.

Also, the extraction of `normalize_uri` doesn't yield more readability than just doing 
`JSON.parse(URI(uri).read)`

**Review by winston — CHANGES_REQUESTED** [review]
The specs are failing?

File: lib/whatsnew/remote_files.rb (hunk 2 above)
Comment (JuanitoFatas) [inline lib/whatsnew/remote_files.rb]: > Also, the extraction of normalize_uri doesn't yield more readability than just doing 
JSON.parse(URI(uri).read)

Well, I was thinking if I want to do more normalizations, and if that is going to happen, there will be the need for putting in a normalize_uri method, but for now I only want to keep it simple.

File: lib/whatsnew/remote_files.rb (hunk 1 above)
Comment (JuanitoFatas) [inline lib/whatsnew/remote_files.rb]: It is part of the coditional.

**Pull request merged**